src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -137,3 +142,36 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
 }
 
 export default ApplicationView
+
+const getDropdownOptions = (
+  options: (ObjectOption | string)[],
+  currentResponse: ResponseFull | null,
+  addedOption: string | null,
+  setAddedOption: (value: string) => void,
+  displayProperty?: string
+) => {
+  // Figure out if the currentResponse contains an added custom response. This
+  // is just for initial loading if we've had to wait for the list of options to
+  // populate
+  const initialCustomResponse =
+    !addedOption &&
+    currentResponse?.optionIndex === options.length &&
+    typeof currentResponse.selection === 'string'
+      ? currentResponse.selection
+      : null
```

Comment:
This is what allows the custom option to persist after a slow options load. When the component first loads, the saved response won't correspond to the options list. But after options loads, if the saved index is one higher than the options list size, then we assume it's meant to be a custom option for the current list.

Reply:
Nice. I'm wondering if the options dynamically changed to a list that was 1 shorter than the previous and the last option on the list was selected, the selected option could be preserved and brought over to the new list. However, that seems like a very unlikely case and no big deal if it does occur.

Reply:
Yeah, exactly. I figured if and when the options list changes from under you, you're going to want to make a new selection anyway, so the current response becomes irrelevant. It's possible it could "auto-select" the extra option, but that problem's always been there -- if the list changes the index of your current selection will be selected in the new list.